scala/private/phases/phase_compile.bzl

```diff
@@ -106,7 +106,7 @@ def _phase_compile_default(ctx, p, _args = struct()):
         ctx,
         p,
         _args.srcjars if hasattr(_args, "srcjars") else depset(),
-        _args.buildijar if hasattr(_args, "buildijar") else True,
+        _args.buildijar if hasattr(_args, "buildijar") else False,
```

Comment:
I think this change can be contributed as a separate PR to rules scala, just do a conditional according scala version. Example:
```
load("@io_bazel_rules_scala_config//:config.bzl", "SCALA_MAJOR_VERSION", "SCALA_VERSION")

buildijar_default_value = True if SCALA_VERSION.startswith("2.") else False
```